core/trino-main/src/main/java/io/trino/server/security/oauth2/OidcDiscoveryConfig.java

```diff
@@ -19,37 +19,131 @@
 
 import javax.validation.constraints.NotNull;
 
+import java.util.Optional;
+
+import static io.trino.server.security.oauth2.StaticOAuth2ServerConfiguration.ACCESS_TOKEN_ISSUER;
+import static io.trino.server.security.oauth2.StaticOAuth2ServerConfiguration.AUTH_URL;
+import static io.trino.server.security.oauth2.StaticOAuth2ServerConfiguration.JWKS_URL;
+import static io.trino.server.security.oauth2.StaticOAuth2ServerConfiguration.TOKEN_URL;
+import static io.trino.server.security.oauth2.StaticOAuth2ServerConfiguration.USERINFO_URL;
 import static java.util.concurrent.TimeUnit.SECONDS;
 
 public class OidcDiscoveryConfig
 {
-    private Duration metadataReadTimeout = new Duration(30, SECONDS);
-    private boolean useUserInfoEndpoint = true;
+    private Duration discoveryTimeout = new Duration(30, SECONDS);
+    private boolean userInfoEndpointEnabled = true;
+
+    // Left for backward compatibility, remove after the next release/a couple of releases
+    private Optional<String> accessTokenIssuer = Optional.empty();
+    private Optional<String> authUrl = Optional.empty();
+    private Optional<String> tokenUrl = Optional.empty();
+    private Optional<String> jwksUrl = Optional.empty();
+    private Optional<String> userinfoUrl = Optional.empty();
 
-    public boolean isUseUserInfoEndpoint()
+    @NotNull
+    public Duration getDiscoveryTimeout()
     {
-        return useUserInfoEndpoint;
+        return discoveryTimeout;
+    }
+
+    @Config("http-server.authentication.oauth2.oidc.discovery.timeout")
+    @ConfigDescription("OpenID Connect discovery timeout")
+    public OidcDiscoveryConfig setDiscoveryTimeout(Duration discoveryTimeout)
+    {
+        this.discoveryTimeout = discoveryTimeout;
+        return this;
+    }
+
+    public boolean isUserInfoEndpointEnabled()
+    {
+        return userInfoEndpointEnabled;
     }
 
     @Config("http-server.authentication.oauth2.oidc.use-userinfo-endpoint")
     @ConfigDescription("Use userinfo endpoint from OpenID connect metadata document")
-    public OidcDiscoveryConfig setUseUserInfoEndpoint(boolean useUserInfoEndpoint)
+    public OidcDiscoveryConfig setUserInfoEndpointEnabled(boolean userInfoEndpointEnabled)
+    {
+        this.userInfoEndpointEnabled = userInfoEndpointEnabled;
+        return this;
+    }
+
+    @NotNull
+    @Deprecated
+    public Optional<String> getAccessTokenIssuer()
```

Comment:
Shouldn't these methods be annotated with `@LegacyConfig` or `@DefunctConfig` instead of `@Deprecated`?

Reply:
If I understand it correctly `@LegacyConfig` is used to mark a configuration property which changed it's name and `@DefunctConfig` is used to mark a property which was removed entirely. If that's the case I think `@Deprecated` makes the sense.

Reply:
At some point we would like to remove these properties.